### DISCUSSION TITLE
Consider re-licensing under an open source license?

### DISCUSSION THREAD

**Discussion by kylewlacy** [MolotovCherry/virtual-display-rs#45] [General]
I'm in need of a virtual display driver for a project I'm working on, and I believe `virtual-display-rs` ticks most of the boxes for what I'm looking for. Unfortunately, the current license (GPL + [Commons Clause](https://commonsclause.com/)) would be in conflict with how I want to license my project (my tentative plan is to distribute the code under the MIT license, but I'd also like to have the option for paid listings in storefronts such as Steam-- similar to the model [Aseprite](https://www.aseprite.org/) uses).

Is there a chance that the current code could be moved to a more permissive license, or at least one that meets the definition of open source (since the Commons Clause is explicitly not open source)? I'd be willing to provide a few code contributes or a GitHub sponsorship for the project if it ends up being something I could use, although since what I'm working on is a side project, I can't commit to giving that much unfortunately.

**Comment by kylewlacy** [comment]
I also just came across the conversation in https://github.com/MolotovCherry/virtual-display-rs/issues/35, which also brought up the subject of re-licensing. Just wanted to include that link in case anyone else comes across this question

**Comment by MolotovCherry** [comment]
Hi! Thanks for your interest in the project!

Regarding the current license, most of my main concerns were about people not contributing back any changes or helping support the project (as well as acknowledgement of course).

I'd definitely love to see any improvements merged back upstream if you do change anything! Every little bit helps make a better project for everyone! (And thanks for being willing, as well as the support, really appreciate that! <3)

I'm still thinking up the best way to approach this (though it's been on the back of my mind as there are so many things to do lately), but as said before, I'm certainly not opposed to opening this up a bit more.

Most likely what I'd open it up to would just be GPLv3 (removing commons clause). Would that be sufficient?

P.S. I looked at your projects a bit, cool stuff! I implemented an emulator once (a simple one since I'm new to it though), it's pretty fun (also hard) !

**Reply by kylewlacy** [reply]
Yes, plain GPLv3 would work for my use case. My app would just integrate with the driver-- meaning the user would install it themselves, then my app would just initialize the device and send messages over the named pipe or whatever (I haven't looked into exactly how the communication works between the userland app and the driver yet... but I imagine it isn't too complicated).

Anyway, my understanding is all of this is compatible with the GPLv3 license, as long as all the other conditions are met (giving the user access to the source code including any modifications I make, letting the user install their own forks of the driver, etc.). So I would definitely be happy to see the project move to a plain GPLv3 license!

**Reply by MolotovCherry** [reply]
I re-licensed it now.
